File: build_index.py
import random

def build():
    # Load original HTML to extract header and footer
    with open("e:\\NoshahiBaba\\index.html", "r", encoding="utf-8") as f:
        content = f.read()
        
    head_end = content.find("</head>")
    header_end = content.find("<main>")
    footer_start = content.find("<footer>")
    
    # We will reconstruct the header nav links exactly as requested
    top_part = content[:header_end]
    # Replace nav links in top_part
    nav_links_start = top_part.find('<div class="nav-links">')
    nav_links_end = top_part.find('</div>', nav_links_start) + 6
    if nav_links_start != -1:
        new_nav_links = '''<div class="nav-links">
                    <a href="index.html" class="active"><i data-lucide="home"></i> Home</a>
                    <a href="featured.html">Featured selections</a>
                </div>'''
        top_part = top_part[:nav_links_start] + new_nav_links + top_part[nav_links_end:]
        
    # Remove right part nav if needed
    nav_right_start = top_part.find('<div class="nav-right">')
    if nav_right_start != -1:
        nav_right_end = top_part.find('</div>', nav_right_start) + 6
        top_part = top_part[:nav_right_start] + top_part[nav_right_end:]

    # CSS for product sections
    css = """
    <style>
        body { background-color: #ffffff !important; color: #333333 !important; }
        .section-divider { border: 0; height: 2px; background: #e0e0e0; margin: 40px auto; max-width: 1200px; }
        
        .product-card, .tp-card, .card-style-1, .card-style-2, .card-style-3, .card-style-4, .card-style-5 {
            transition: none !important;
            transform: none !important;
            animation: none !important;
            box-shadow: none !important; /* ensure no hover box shadow animation */
        }
        
        .prod-grid {
            display: grid;
            grid-template-columns: repeat(auto-fill, minmax(220px, 1fr));
            gap: 20px;
            padding: 20px 0;
            max-width: 1200px;
            margin: 0 auto;
        }
        
        .section-title {
            text-align: center;
            font-size: 32px;
            margin-bottom: 20px;
            color: #111;
            font-weight: 700;
        }
        
        .img-box {
            width: 100%;
            height: 200px;
            background: #fff;
            position: relative;
            overflow: hidden;
            border-bottom: 1px solid #eee;
        }
        
        .img-box img {
            width: 100%;
            height: 100%;
            object-fit: contain;
        }
        
        /* Full screen image on hover */
        .card-base:hover .img-box img {
            position: fixed;
            top: 0;
            left: 0;
            width: 100vw;
            height: 100vh;
            z-index: 99999;
            background-color: rgba(255, 255, 255, 0.95);
            object-fit: contain;
            padding: 40px;
            box-sizing: border-box;
            backdrop-filter: blur(5px);
        }
        
        /* Styles */
        .card-style-1 { border: 1px solid #ddd; border-radius: 8px; background: #fff; padding-bottom: 15px; }
        .card-style-1 .prod-info { padding: 10px; }
        
        .card-style-2 { border: 2px solid #f0f0f0; background: #fff; border-radius: 4px; }
        .card-style-2 .prod-info { padding: 15px; text-align: center; }
        
        .card-style-3 { border: 1px solid #333; background: #fafafa; }
        .card-style-3 .prod-info { padding: 12px; }
        
        .card-style-4 { border-radius: 20px; border: 1px solid #eee; background: #fff; border-bottom: 4px solid #FFCC00; overflow: hidden; }
        .card-style-4 .prod-info { padding: 15px; }
        
        .card-style-5 { border: 1px solid #e0e0e0; background: #fff; position: relative; }
        .card-style-5 .prod-info { padding: 15px; background: #fcfcfc; }
        
        .price { color: #d9534f; font-weight: 700; font-size: 18px; margin: 5px 0; }
        .title { font-size: 14px; font-weight: 500; color: #333; line-height: 1.4; height: 40px; overflow: hidden; }
        .supplier-badge { color: #666; font-size: 12px; display: flex; align-items: center; gap: 5px; margin-top: 10px; }
        
        /* Top pick adjustments to make it look responsive */
        @media (max-width: 768px) {
            .prod-grid { grid-template-columns: repeat(2, 1fr); gap: 10px; }
            .img-box { height: 150px; }
            .section-title { font-size: 24px; }
        }
    </style>
    """
    
    # Inject CSS before </head>
    if head_end != -1:
        top_part = top_part[:head_end] + css + top_part[head_end:]

    main_start = '    <main class="container" style="padding-top: 40px;">\n'
    
    section_names = ["Just for You", "Featured Innovations", "Trending Products", "Top Rated Accessories", "New Arrivals"]
    image_urls = [
        "https://images.unsplash.com/photo-1546435770-a3e426bf472b?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1590658268037-6bf12165a8df?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1505740420928-5e560c06d30e?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1523275335684-37898b6baf30?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1583394838336-acd977736f90?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1572569438068-4099edbf738e?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1560393464-5c69a73c5770?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1526170375885-4d8ecf77b99f?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1542291026-7eec264c27ff?q=80&w=400&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1627989580309-bfaf3e58af6f?q=80&w=400&auto=format&fit=crop",
    ]

    sections_html = ""
    for i, section in enumerate(section_names):
        style_class = f"card-style-{i+1}"
        sections_html += f'        <section>\n'
        sections_html += f'            <h2 class="section-title">{section}</h2>\n'
        sections_html += f'            <div class="prod-grid">\n'
        
        for j in range(20):
            img = random.choice(image_urls)
            price = random.randint(15, 250)
            sections_html += f'''                <div class="{style_class} card-base">
                    <div class="img-box">
                        <img src="{img}" alt="Product {j}">
                    </div>
                    <div class="prod-info">
                        <div class="title">Premium Quality Product {j+1} - Great Design</div>
                        <div class="price">PKR {price},000</div>
                        <div class="supplier-badge"><i data-lucide="shield-check"></i> Verified Supplier</div>
                    </div>
                </div>\n'''
                    
        sections_html += f'            </div>\n'
        sections_html += f'        </section>\n'
        if i < len(section_names) - 1:
            sections_html += f'        <hr class="section-divider">\n'

    main_end = '\n    </main>\n'
    
    footer_part = content[footer_start:]
    
    with open("e:\\NoshahiBaba\\index.html", "w", encoding="utf-8") as f:
        f.write(top_part + main_start + sections_html + main_end + footer_part)

if __name__ == "__main__":
    build()

File: build_index.js
const fs = require('fs');
const path = require('path');

function build() {
    const filePath = path.join('e:', 'NoshahiBaba', 'index.html');
    const content = fs.readFileSync(filePath, 'utf-8');

    const headEnd = content.indexOf('</head>');

    // Find the product grid start
    const productGridStart = content.indexOf('<!-- Product Grid -->');
    let topPart = content.substring(0, productGridStart);

    // Remove the extra links from the navbar in topPart
    // We want to keep "All categories", "Home", "Featured selections"
    // "All categories" is its own div outside .nav-links.
    // So we just replace the inner nav links and remove nav-right.
    const navLinksStart = topPart.indexOf('<div class="nav-links">');
    if (navLinksStart !== -1) {
        let navLinksEnd = topPart.indexOf('</div>', navLinksStart) + 6;
        const newNavLinks = '<div class="nav-links">\n                    <a href="index.html" class="active"><i data-lucide="home"></i> Home</a>\n                    <a href="featured.html">Featured selections</a>\n                </div>';
        topPart = topPart.substring(0, navLinksStart) + newNavLinks + topPart.substring(navLinksEnd);
    }

    const navRightStart = topPart.indexOf('<div class="nav-right">');
    if (navRightStart !== -1) {
        let navRightEnd = topPart.indexOf('</div>', navRightStart) + 6;
        topPart = topPart.substring(0, navRightStart) + topPart.substring(navRightEnd);
    }

    const css = `<style>
        body { background-color: #ffffff !important; color: #333333 !important; }
        .section-divider { border: 0; height: 1px; background: #e0e0e0; margin: 50px auto; max-width: 1400px; }
        .card-base { transition: transform 0.2s, box-shadow 0.2s !important; cursor: pointer; }
        .card-base:hover { transform: translateY(-3px) !important; box-shadow: 0 8px 25px rgba(0,0,0,0.1) !important; }
        
        .prod-grid { display: grid; grid-template-columns: repeat(auto-fill, minmax(280px, 1fr)); gap: 30px; padding: 20px 0; max-width: 1400px; margin: 0 auto; }
        
        .section-title-wrapper { text-align: center; margin-bottom: 40px; margin-top: 20px; }
        .section-title-wrapper .section-title { font-size: 34px; color: #111; font-weight: 800; position: relative; display: inline-block; padding-bottom: 15px; margin: 0;}
        .section-title-wrapper .section-title::after { content: ''; position: absolute; bottom: 0; left: 50%; transform: translateX(-50%); width: 80px; height: 5px; background: #FFCC00; border-radius: 5px; }
        
        .img-box { width: 100%; height: 240px; background: #f9f9f9; position: relative; overflow: hidden; border-bottom: 1px solid #eee; display: flex; align-items: center; justify-content: center; }
        .img-box img { width: 100%; height: 100%; object-fit: contain; padding: 10px; }
        
        /* Modal Styles */
        .modal-overlay { display: none; position: fixed; top: 0; left: 0; width: 100vw; height: 100vh; background: rgba(0,0,0,0.85); z-index: 99999; align-items: center; justify-content: center; backdrop-filter: blur(8px); }
        .modal-overlay.active { display: flex; }
        .full-card { background: white; width: 90%; max-width: 550px; border-radius: 16px; padding: 30px; position: relative; box-shadow: 0 25px 60px rgba(0,0,0,0.3); }
        .close-modal { position: absolute; top: -15px; right: -15px; background: #fff; border: 2px solid #ccc; width: 40px; height: 40px; border-radius: 50%; cursor: pointer; font-size: 24px; font-weight: bold; display: flex; align-items: center; justify-content: center; transition: all 0.2s; z-index: 2; box-shadow: 0 4px 10px rgba(0,0,0,0.2); color: #333; }
        .close-modal:hover { background: #FFCC00; border-color: #FFCC00; color: #000; transform: scale(1.1); }
        
        /* Modal specific sizing overrides */
        #modalInner .img-box { height: 350px; background: #fff; border:none; margin-bottom: 20px; border-radius: 12px; }
        #modalInner .img-box img { padding: 0; }
        #modalInner .title { font-size: 22px; height: auto; font-weight: 700; margin-bottom: 15px; line-height: 1.3;}
        #modalInner .price { font-size: 26px; margin-bottom: 15px; }
        #modalInner .source-btn { padding: 18px; font-size: 18px; margin-top: 25px; }
        #modalInner .supplier-badge { font-size: 14px; }
        
        /* Card variations */
        .card-style-1 { border: 1px solid #e5e5e5; border-radius: 12px; background: #fff; padding-bottom: 5px; overflow: hidden; }
        .card-style-1 .prod-info { padding: 20px; }
        
        .card-style-2 { border: none; background: #fff; border-radius: 12px; box-shadow: 0 4px 15px rgba(0,0,0,0.06); overflow: hidden; padding-bottom: 5px;}
        .card-style-2 .prod-info { padding: 20px; text-align: center; }
        
        .card-style-3 { border: 1px solid #ddd; background: #fafafa; border-radius: 8px; padding-bottom: 5px;}
        .card-style-3 .prod-info { padding: 20px; }
        
        .card-style-4 { border-radius: 16px; border: 1px solid #eee; background: #fff; border-bottom: 5px solid #FFCC00; overflow: hidden; padding-bottom: 5px;}
        .card-style-4 .prod-info { padding: 20px; }
        
        .card-style-5 { border: 1px solid #eee; background: #ffffff; position: relative; border-left: 5px solid #FFCC00; border-radius: 8px; padding-bottom: 5px;}
        .card-style-5 .prod-info { padding: 20px; }
        
        .card-base .price { color: #d9534f; font-weight: 800; font-size: 20px; margin: 8px 0; }
        .card-base .title { font-size: 16px; font-weight: 600; color: #222; line-height: 1.4; height: 44px; overflow: hidden; }
        .card-base .supplier-badge { color: #555; font-size: 13px; display: flex; align-items: center; gap: 6px; margin-top: 12px; font-weight: 500;}
        .card-base .supplier-badge svg { color: #FFCC00; } /* Yellow accent for icon */
        
        .card-base .source-btn { width: 100%; padding: 12px; background: #FFCC00; color: #000; font-weight: 700; border: none; border-radius: 8px; margin-top: 18px; cursor: pointer; transition: all 0.2s; font-size: 15px; }
        .card-base .source-btn:hover { background: #e6b800; box-shadow: 0 4px 12px rgba(255, 204, 0, 0.4); }
        
        @media (max-width: 768px) { .prod-grid { grid-template-columns: repeat(2, 1fr); gap: 15px; padding: 10px; } .img-box { height: 180px; } .section-title-wrapper .section-title { font-size: 26px; } }
        @media (max-width: 480px) { .prod-grid { grid-template-columns: 1fr; } }
        </style>
    `;

    // Inject the CSS at the right spot
    const styleEndIdx = topPart.indexOf('</head>');
    if (styleEndIdx !== -1) {
        topPart = topPart.substring(0, styleEndIdx) + css + topPart.substring(styleEndIdx);
    }

    const sectionNames = ["Just for You", "Featured Innovations", "Trending Products", "Top Rated Accessories", "New Arrivals"];
    const imageUrls = [
        "https://images.unsplash.com/photo-1546435770-a3e426bf472b?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1590658268037-6bf12165a8df?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1505740420928-5e560c06d30e?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1523275335684-37898b6baf30?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1583394838336-acd977736f90?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1572569438068-4099edbf738e?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1560393464-5c69a73c5770?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1526170375885-4d8ecf77b99f?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1542291026-7eec264c27ff?q=80&w=600&auto=format&fit=crop",
        "https://images.unsplash.com/photo-1627989580309-bfaf3e58af6f?q=80&w=600&auto=format&fit=crop"
    ];

    let sectionsHtml = "";
    for (let i = 0; i < sectionNames.length; i++) {
        const styleClass = "card-style-" + (i + 1);
        sectionsHtml += "        <section>\n";
        sectionsHtml += "            <div class=\"section-title-wrapper container\">\n";
        sectionsHtml += "                <h2 class=\"section-title\">" + sectionNames[i] + "</h2>\n";
        sectionsHtml += "            </div>\n";
        sectionsHtml += "            <div class=\"prod-grid container\">\n";

        for (let j = 0; j < 20; j++) {
            const img = imageUrls[Math.floor(Math.random() * imageUrls.length)];
            const price = Math.floor(Math.random() * 235) + 15;
            sectionsHtml += "                <div class=\"" + styleClass + " card-base\" onclick=\"openModal(this)\">\n" +
                "                    <div class=\"img-box\">\n" +
                "                        <img src=\"" + img + "\" alt=\"Product " + j + "\">\n" +
                "                    </div>\n" +
                "                    <div class=\"prod-info\">\n" +
                "                        <div class=\"title\">Premium " + sectionNames[i] + " Item " + (j + 1) + " - Professional Quality Design</div>\n" +
                "                        <div class=\"price\">PKR " + price + ",000</div>\n" +
                "                        <div class=\"supplier-badge\"><i data-lucide=\"shield-check\"></i> Verified Supplier</div>\n" +
                "                        <button class=\"source-btn\">Source Now</button>\n" +
                "                    </div>\n" +
                "                </div>\n";
        }
        sectionsHtml += "            </div>\n";
        sectionsHtml += "        </section>\n";
        if (i < sectionNames.length - 1) {
            sectionsHtml += "        <hr class=\"section-divider\">\n";
        }
    }

    const modalHtml = `
    <!-- Modal Overlay -->
    <div class="modal-overlay" id="cardModal" onclick="closeModal(event)">
        <div class="full-card" id="modalContent" onclick="event.stopPropagation()">
            <button class="close-modal" onclick="closeModal(event)">&times;</button>
            <div id="modalInner"></div>
        </div>
    </div>
    <script>
        function openModal(el) {
            document.getElementById('modalInner').innerHTML = el.innerHTML;
            document.getElementById('cardModal').classList.add('active');
            // Ensure icons are re-rendered in the modal if needed
            if (window.lucide) {
                lucide.createIcons({
                    root: document.getElementById('modalInner')
                });
            }
        }
        function closeModal(e) {
            document.getElementById('cardModal').classList.remove('active');
        }
    </script>
    `;

    // Find the end of main to extract footer
    const mainEndIdx = content.indexOf('</main>');
    const bottomPart = content.substring(mainEndIdx);

    fs.writeFileSync(filePath, topPart + sectionsHtml + modalHtml + '\n' + bottomPart, 'utf-8');
}

build();
